test/support/utils.rb

```diff
@@ -91,6 +89,22 @@ def create_pseudo_terminal(boot_options: "-r debug/run")
 
     private
 
+    # use this to start a debug session with the test program
+    def start_debugger_with(program)
+      write_temp_file(strip_line_num(program))
+      inject_lib_to_load_path
+      pid = spawn("#{RUBY} -r debug/open #{temp_file_path}")
+      sleep(0.5) # wait for the debug server to start
+      require_relative "../../lib/debug/client"
+      connect
```

Comment:
This is my task but `client.rb` is not expected, so I need to change it to `DEBUGGER__.connect` or something similar.

Reply:
Another  comment is, if there is another remote debug session, it will fail, so `debug/open` with specific path file `RUBY_DEBUG_SOCK_PATH` and connect to it is good idea.

Reply:
thanks for the hint. I'll update it 👍